### ISSUE TITLE
Loss does not decrease when training `nemo_automodel.components.models.gpt2.build_gpt2_model` without a base model due to missing `initialize_weights` definition

### ISSUE THREAD

**Issue by fushigu** [NVIDIA-NeMo/Automodel#1071]
**Describe the bug**

When training `nemo_automodel.components.models.gpt2.build_gpt2_model` without a base model following the tutorial, the loss does not decrease.

When `nemo_automodel.components.models.gpt2.build_gpt2_model` is executed, weight initialization is not performed, and `self.wte` inside the `GPT2LMHeadModel` module ends up being initialized to zeros.
(`GPT2LMHeadModel` is defined at `Automodel/nemo_automodel/components/models/gpt2.py`)

The root cause appears to be that `GPT2LMHeadModel` does not define an `initialize_weights` method.

**Steps / Code to reproduce bug**

Use the configuration below and run:
```
uv run torchrun --nproc-per-node=1 examples/llm_pretrain/pretrain.py -c /path/to/config.yaml
```
```
# Example configuration for pretraining a nanoGPT-style model on binary token shards
# produced by the `modded-nanogpt` preprocessing pipeline.

step_scheduler:
  global_batch_size: 32
  local_batch_size: 2
  ckpt_every_steps: 2000
  num_epochs: 1
  max_steps: 9500

dist_env:
  backend: nccl
  timeout_minutes: 1

# Model
# Option 1 – shorthand builder
model:
  _target_: nemo_automodel.components.models.gpt2.build_gpt2_model
  vocab_size: 50258   # default 50257
  n_positions: 2048   # context length
  n_embd: 768         # hidden size
  n_layer: 12         # number of transformer blocks
  n_head: 12         # attention heads
  bos_token_id: 50256
  eos_token_id: 50256

# Option 2 – explicit HF config (uncomment to use)
# model:
#   _target_: nemo_automodel.NeMoAutoModelForCausalLM.from_config
#   config:
#     _target_: transformers.GPT2Config
#     vocab_size: 50258
#     n_positions: 2048
#     n_ctx: 2048
#     n_embd: 768
#     n_layer: 12
#     n_head: 12
#     bos_token_id: 50256
#     eos_token_id: 50256

# Dataset
dataset:
  _target_: nemo_automodel.components.datasets.llm.nanogpt_dataset.NanogptDataset
  file_pattern: "/path/to/fineweb_max_tokens_500M/dataset.bin"
  seq_len: 1024
  shuffle_files: true
  align_to_bos: false
  bos_token: 50256

dataloader:
  _target_: torchdata.stateful_dataloader.StatefulDataLoader
  shuffle: false
  collate_fn: nemo_automodel.components.datasets.utils.default_collater

# Loss
loss_fn:
  _target_: nemo_automodel.components.loss.masked_ce.MaskedCrossEntropy

# Optimizer
optimizer:
  _target_: torch.optim.AdamW
  lr: 2.0e-4
  betas: [0.9, 0.95]
  weight_decay: 0.1

# Distributed
distributed:
  _target_: nemo_automodel.components.distributed.fsdp2.FSDP2Manager
  dp_size: none
  tp_size: 1
  cp_size: 1
  sequence_parallel: false

checkpoint:
  enabled: true
  checkpoint_dir: checkpoints/
  model_save_format: torch_save
  load_base_model: false
```

**Expected behavior**

At `self.wte` should be initialized via `initialize_weights`, and set to non-zero values.

**Additional context**

Adding the following method to `GPT2LMHeadModel` results in proper weight initialization:

```
def initialize_weights(self):
    self._init_weights()
```

**Comment by qiaochuz-nv** [comment]
## QA RCCA Analysis
**Fix PR:** Not identified
**Root Cause:** GPT2 custom model built via `from_config` (no pretrained weights). `_init_weights` called in constructor may conflict with meta-device initialization, causing loss to stay flat.
**Trigger Config:** Custom GPT2 model via `build_gpt2_model` + training from scratch
**Functional Test Coverage: NEW TEST CREATED** — GPT2 from_config training-from-scratch test added.
**Conclusion:** Test created to cover from_config training path.
